### PR TITLE
ARROW-5934: [Python] Bundle arrow's LICENSE with the wheels

### PR DESCRIPTION
**Pull request by kszucs**
Setuptools can handle multiple license files, but they get overwritten because of the same naming, so move the content python/LICENSE.txt to the top level LICENSE.txt.  I've investigated a solution to concatenate these, but it would require to extend multiple distutils/setuptools/wheel commands in a really error-prone fashion.
- Distribute the top level LICENSE.txt file with the wheels 
- Distribute the top level LICENSE.txt with the source distribution, however `python setup.py bdist` leaves LICENSE.txt file as a garbage in the rebository, it cannot cleanup properly probably because the relative directory reference.

Add 3rdparty licenses because of the static-linkage in the wheels:
- lz4: https://github.com/lz4/lz4/blob/dev/lib/LICENSE#L11-L13
- zstd: https://github.com/facebook/zstd/blob/dev/LICENSE#L13-L15
- rapidjson: because of transitive dependencies https://github.com/Tencent/rapidjson/blob/master/license.txt#L21
- uriparser: https://github.com/uriparser/uriparser/blob/master/COPYING#L15-L18
- double-conversion: https://github.com/google/double-conversion/blob/3f9cd30e1bca91c0a036ad8b2b7eb8e2d2290dd2/LICENSE#L8-L11
- snappy: https://github.com/google/snappy/blob/master/COPYING#L10-L13
- brotli: https://github.com/google/brotli/blob/master/LICENSE#L10
- jemalloc is already added
- protobuf is already added
- gflags: https://github.com/gflags/gflags/blob/master/COPYING.txt#L10-L13
- c-ares: https://github.com/c-ares/c-ares/blob/master/LICENSE.md
- glog: https://github.com/google/glog/blob/master/COPYING#L10-L13
- boost is already added
- flatbuffers is already added
- grpc: https://github.com/grpc/grpc/blob/master/NOTICE.txt
- gtest: do we redistribute any binaries from it?
- gbenchmark: do we redistribute any binaries from it?
- apache thrift added
- apache orc added
- zlip is redistributed as a shared library with all wheels (in the future we should switch to static linkage)
- openssl and libcrypto is redistributed as shared libraries with the windows wheels, so added the openssl license preceding version 3 

bzip2 doesn't require to include its license with binary redistributions: https://github.com/asimonov-im/bzip2/blob/master/LICENSE

### CODE REVIEW

**Comment by kszucs** [comment]
cc @wesm @xhochy @nealrichardson

**Comment by kszucs** [comment]
@xhochy what else do we statically link in the wheels?

File: LICENSE.txt [hunk 1]
```diff
@@ -936,6 +937,732 @@ support library is itself covered by the above license.
 
 --------------------------------------------------------------------------------
 
+3rdparty dependency gRPC is statically linked in certain binary
```
Comment (wesm) [inline LICENSE.txt]: What do you think about putting each license in its own file in a licenses directory like Apache Spark? This is easier to manage and also makes the binary licenses more visible

https://github.com/apache/spark/tree/master/licenses-binary

I think we should probably also start managing the licenses for thirdparty source inclusions in their own directory like Apache Spark.

Comment (kszucs) [reply LICENSE.txt]: That is indeed nicer and more manageable! How about separating them for 1.0? I'd like to cut RC0 as soon as possible.

**Review by wesm — APPROVED** [review]
+1

File: LICENSE.txt (hunk 1 above)
Comment (wesm) [reply LICENSE.txt]: Sure thing

Comment (kszucs) [reply LICENSE.txt]: Created issue for it https://issues.apache.org/jira/browse/ARROW-5954

**Pull request closed without merging**